services/motion/builtin/move_request.go

```diff
@@ -270,24 +237,32 @@ func (mr *moveRequest) obstaclesIntersectPlan(
 				return state.ExecuteResponse{}, err
 			}
 
+			// build representation of frame system's inputs
 			currentInputs, err := mr.kinematicBase.CurrentInputs(ctx)
 			if err != nil {
 				return state.ExecuteResponse{}, err
 			}
+			inputMap := referenceframe.StartPositions(mr.planRequest.FrameSystem)
+			inputMap[mr.kinematicBase.Name().ShortName()] = currentInputs
 
 			// get the pose difference between where the robot is versus where it ought to be.
-			errorState, err := mr.kinematicBase.ErrorState(ctx, waypoints, waypointIndex)
+			errorState, err := mr.kinematicBase.ErrorState(ctx, plan, waypointIndex)
 			if err != nil {
 				return state.ExecuteResponse{}, err
 			}
 
+			// check no obstacles intersect the portion of the plan which has yet to be executed
+			remainingPlan, err := motionplan.RemainingPlan(plan, waypointIndex)
```

Comment:
i don't think this should be here. 
Even if waypointIndex gets incremented I do not think this would recognize that change since `waypointIndex` is of type `float`

Reply:
I don't understand this comment

Reply:
i think us calculating the remaining plan should happen outside the for loop. 
my understanding is that remaining plan will always be the same once we enter this function.

Reply:
Ah I see.  Yes I think this is an improvement.  I will make this change

Reply:
Changed